tests/e2e/mod.rs

```diff
@@ -145,8 +148,23 @@ fn retail_supergraph(clone_retail_supergraph_repo: &'static TempDir) -> RetailSu
     }
 }
 
+struct SingleMutableSubgraph {
+    subgraph_url: String,
+    directory: TempDir,
+    schema_file_name: String,
+    task_handle: Child,
+}
+
+impl Drop for SingleMutableSubgraph {
+    fn drop(&mut self) {
+        self.task_handle
+            .start_kill()
+            .expect("Could not kill underlying task");
+    }
+}
```

Comment:
I once read being able to panic in a drop could produce UB - not sure if this is still valid, but maybe this could be a `stop()` method that we have to call manually?

Reply:
Actually I think this is vestigial and left over from an old version of the PR, I can just tell the handle to kill on drop anyway and then we avoid this issue.